src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -29,6 +31,20 @@
 )
 
 
+class _IntegratorInfo(TypedDict):
```

Comment:
Hm, I see why you wanted to make these classes. You could do that, and just have each have an `integrate` method? Just as a way to store this information more naturally? 

Otherwise I'd say put this in the file with the push algorithm, but the class idea is better

Reply:
Yeah the class idea is definitely the way to go here. Otherwise the typing gets a little dubious because Python's typing for `Callable` is very messy https://docs.python.org/3/library/typing.html#annotating-callable-objects

Reply:
Ok sounds great! Maybe you can prioritize these changes to this PR today and we can try to get this in today/early next week? Looks pretty close to done to me.

Reply:
So I have it implemented but I'm not really happy with how I handle `is_relativistic`: https://github.com/PlasmaPy/PlasmaPy/blob/c6aee176c601687992d25e50684ad2936ff33a82/src/plasmapy/simulation/particle_integrators.py#L24-L28

I need a clean way of indicating that this property has to be defined by specifying it in the abstract class somewhere while also not requiring the class to be instantiated-- something like an "abstract property"

Reply:
Does abstract property from ABC not do what you want? https://docs.python.org/3/library/abc.html#abc.abstractproperty

Reply:
It does, but part of the issue is that we shouldn't be requiring the user to instantiate the integrator (which is required with abstract properties.) I might just stick with the `TypedDict` approach and expose the `AbstractIntegrator` as such a dict

Reply:
I had assumed the classes would all be defined here, then in tracker the function would take a keyword argument to specify which tracker to use, then instantiate it internally, `self._integrator = BorisIntegrator()` then use that for the push function? Maybe I don't understand the issue here?

The documentation is a good thing to think of: at least in this module, we want documentation for each integrator method which the classes seem a natural match for.

Reply:
I'll switch it over to instantiate the integrator, definitely makes it easier to ensure the relevant documentation is included